cmd/run.go

```diff
@@ -435,19 +438,438 @@ func init() {
+			err = utils.IsDaprListeningOnPort(runE.DaprHTTPPort, time.Duration(runtimeWaitTimeoutInSeconds)*time.Second)
+			if err != nil {
+				sidecarUp = false
+				print.StatusEvent(runE.DaprCMD.OutputWriter, print.LogWarning, "Dapr sidecar is not listening on HTTP port: %s", err.Error())
+			}
+
+			print.StatusEvent(runE.DaprCMD.OutputWriter, print.LogInfo, "Checking if Dapr sidecar is listening on GRPC port %v", runE.DaprGRPCPort)
+			err = utils.IsDaprListeningOnPort(runE.DaprGRPCPort, time.Duration(runtimeWaitTimeoutInSeconds)*time.Second)
+			if err != nil {
+				sidecarUp = false
+				print.StatusEvent(runE.DaprCMD.OutputWriter, print.LogWarning, "Dapr sidecar is not listening on GRPC port: %s", err.Error())
+			}
+		}
+
+		if sidecarUp {
+			print.StatusEvent(runE.DaprCMD.OutputWriter, print.LogInfo, "Dapr sidecar is up and running.")
+		} else {
+			print.StatusEvent(runE.DaprCMD.OutputWriter, print.LogWarning, "Dapr sidecar might not be responding.")
+		}
+	}
+	daprRunning <- true
+}
+
+// killDaprdProcess is used to kill the Daprd process and return error on failure.
+func killDaprdProcess(runE *runExec.RunExec) error {
+	err := runE.DaprCMD.Command.Process.Kill()
+	if err != nil {
+		print.StatusEvent(runE.DaprCMD.ErrorWriter, print.LogFailure, "Error exiting Dapr: %s", err)
+		return err
+	}
+	print.StatusEvent(runE.DaprCMD.OutputWriter, print.LogSuccess, "Exited Dapr successfully")
+	return nil
+}
+
+// killAppProcess is used to kill the App process and return error on failure.
+func killAppProcess(runE *runExec.RunExec) error {
+	err := runE.AppCMD.Command.Process.Kill()
+	if err != nil {
+		print.StatusEvent(runE.DaprCMD.ErrorWriter, print.LogFailure, "Error exiting App: %s", err)
+		return err
+	}
+	print.StatusEvent(runE.DaprCMD.OutputWriter, print.LogSuccess, "Exited App successfully")
+	return nil
+}
+
+// putCLIProcessIDInMeta puts the CLI process ID in metadata so that it can be used by the CLI to stop the CLI process.
+func putCLIProcessIDInMeta(runE *runExec.RunExec, pid int) error {
+	// For now putting this as 0, since we do not want the dapr stop command for a single to kill the CLI process,
+	// thereby killing all the apps that are running via dapr run -f.
+	err := metadata.Put(runE.DaprHTTPPort, "cliPID", strconv.Itoa(pid), runE.AppID, unixDomainSocket)
+	if err != nil {
+		print.StatusEvent(runE.DaprCMD.OutputWriter, print.LogWarning, "Could not update sidecar metadata for cliPID: %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+// putAppCommandInMeta puts the app command in metadata so that it can be used by the CLI to stop the app.
+func putAppCommandInMeta(runConfig standalone.RunConfig, runState *runExec.RunExec) error {
+	appCommand := strings.Join(runConfig.Command, " ")
+	print.StatusEvent(runState.DaprCMD.OutputWriter, print.LogInfo, "Updating metadata for app command: %s", appCommand)
+	err := metadata.Put(runState.DaprHTTPPort, "appCommand", appCommand, runState.AppID, runConfig.UnixDomainSocket)
```

Comment:
Would it also make sense to put the run file used as a separate piece of metadata (e.g. as proposed in #1153)?  Else tools are required to parse the `daprd` command line which is indirect and subject to change.  In any case, if the run file contains paths relative to the current directory in which the command was run, the current directory would also need to be in the metadata so that tooling can recreate paths.

Reply:
@philliphoff we are planning to include the runfile path in extended metadata. Please check this link for more discussion - https://github.com/dapr/cli/issues/1165#issuecomment-1386568793